airbyte-integrations/connectors/source-mongodb/lib/mongodb_reader.rb

```diff
@@ -0,0 +1,81 @@
+require_relative './airbyte_protocol.rb'
+require_relative './airbyte_logger.rb'
+
+require_relative './mongodb_stream.rb'
+
+class MongodbReader
+  BATCH_SIZE = 10_000
+  LOG_BATCH_SIZE = 10_000
+
+  def initialize(client:, catalog:, state:)
+    @client = client
+    @catalog = catalog
+    @state = state
+  end
+
+  def read
+    @catalog['streams'].each do |stream_object|
```

Comment:
a better name would be `configured_stream` since that is the name used in the JSON schema

Reply:
In fact `ConfiguredAirbyteCatalog` has a single `streams` key which is an array of `ConfiguredAirbyteStream` objects.